src/sdk/namespace_cache.js

```diff
@@ -401,6 +407,13 @@ class NamespaceCache {
 
 
     async read_object_stream(params, object_sdk) {
+        if (params.part_number) {
```

Comment:
What about part_number===0? I think valid multipart part_number starts from 1 but even then its better to add an explicit comment that will not be confusing to read.